src/main/java/org/jabref/gui/ai/components/aichat/AiChatComponent.java

```diff
@@ -201,9 +208,10 @@ private void addError(String message) {
     private void deleteMessage(ChatMessageComponent chatMessageComponent) {
         blockScroll.set(2);
 
-        aiChatLogic.getChatHistory().remove(chatMessageComponent.getChatMessage());
-
+        int index = chatVBox.getChildren().indexOf(chatMessageComponent);
```

Comment:
This is black magic that this works. Even after deletion of a chat message, removal of others works, too. Even for a restart. -- That means, even if there is a "hole" in the index number, it works.

Do you add invisible children?

Reply:
No, it's not a black magic :), everything is simple, I'm just iterating over entries

Reply:
And then, if I need to find a message at an index, I do `entrySet().stream().skip(index).findFirst()`